docs/user/using-an-OIDC-provider.md

```diff
@@ -59,39 +61,80 @@ In the case of Google we can use an OAuth 2.0 client ID. You can find more infor
+The main difference in the authentication is that instead of accessing the Kubeapps service, we will be accessing an oauth2 proxy service that is in charge of authenticating users with the identity provider and injecting the required credentials in the requests to Kubeapps. There are a number of available solutions for this use-case, like [keycloak-gatekeeper](https://github.com/keycloak/keycloak-gatekeeper) and [oauth2_proxy](https://github.com/pusher/oauth2_proxy). For this guide we will use `oauth2_proxy` since it supports both OIDC and plain OAuth2 for many providers.
 
-Once the proxy is accessible, you will be redirected to your IdP. There, you will be prompted to introduce your credentials:
-
-![Proxy Login](../img/auth-proxy-login.png)
-
-After successfully logging in with your IdP, you will be redirected to Kubeapps and be authenticated with your user's OIDC token.
+Once the proxy is accessible, you will be redirected to the identity provider to authenticate. After successfully authenticating, you will be redirected to Kubeapps and be authenticated with your user's OIDC token.
 
 The next sections explain how you can deploy this proxy either using the Kubeapps chart or manually.
 
 ### Using the chart
 
-Kubeapps chart allows you to automatically deploy the proxy for you as a sidecar container if you specify the necessary flags. In a nutshell you need to enable the feature and set the client ID, secret and the IdP URL. This is an example for using Google as IdP, modify the flags below to adapt them:
+Kubeapps chart allows you to automatically deploy the proxy for you as a sidecar container if you specify the necessary flags. In a nutshell you need to enable the feature and set the client ID, secret and the IdP URL. The following examples use Google as the Identity Provider, modify the flags below to adapt them:
+
+**Example 1: Using the OIDC provider**
+
+This example uses `oauth2-proxy`'s generic OIDC provider with Google, but is applicable to any OIDC provider. Note that the issuer url is passed as an additional flag here, together with an option to enable the cookie being set over an insecure connection for local development only:
+
+```
+helm install bitnami/kubeapps \
+  --namespace kubeapps --name kubeapps \
+  --set authProxy.enabled=true \
+  --set authProxy.provider=oidc \
+  --set authProxy.clientID=my-client-id.apps.googleusercontent.com \
+  --set authProxy.clientSecret=my-client-secret \
+  --set authProxy.cookieSecret=$(echo "not-good-secret" | base64) \
+  --set authProxy.additionalFlags="{-cookie-secure=false,-oidc-issuer-url=https://accounts.google.com}" \
+```
+
+**Example 2: Using a custom provider**
```

Comment:
I do not understand what a custom provider is nor why I should care about them. Are these pre-configured OIDC Idp and that's why you do not need scopes or the `oidc-issuer-urls`? Where can I find more information about them? 

When you say "**Example 2: Using a custom provider** in the section header, are we still talking about  complaint OIDC Id provider?

Reply:
> I do not understand what a custom provider is nor why I should care about them.

I've updated the example title to make it clear that they are oauth2-proxy providers (it is also mentioned in the next sentence), and I had already tried to highlight below why you should care in the following para?

> Are these pre-configured OIDC Idp and that's why you do not need scopes or the oidc-issuer-urls?

Yes. That is, oauth2-proxy provides a bunch of *oauth2* providers, a few of which are actually using openid connect so can be used here if they provide better integration (eg., example 3).

> Where can I find more information about them?

About the providers themselves? I've updated with a link to the oauth2 providers, but it doesn't include details about which providers are using openid connect.